### ISSUE TITLE
build failed at v0.8.13 and lastet master

### ISSUE THREAD

**Issue by smileusd** [kubernetes/node-problem-detector#752]
I just clone the latest code and go get ./... then run make but failed for build bianry:
```
CGO_ENABLED=1 GOOS=linux GO111MODULE=on go build \
	-mod vendor \
	-o bin/node-problem-detector \
	-ldflags '-X k8s.io/node-problem-detector/pkg/version.version=v0.8.13-4-g9fd58e3' \
	-tags "journald " \
	./cmd/nodeproblemdetector
# github.com/coreos/go-systemd/sdjournal
vendor/github.com/coreos/go-systemd/sdjournal/journal.go:27:11: fatal error: systemd/sd-journal.h: No such file or directory
   27 | // #include <systemd/sd-journal.h>
      |           ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:199: bin/node-problem-detector] Error 2
```

**Comment by smileusd** [comment]
There is the full info:
```rm -rf bin/
rm -rf test/bin/
rm -f node-problem-detector-*.tar.gz*
rm -rf output/
rm -f coverage.out
docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
Setting /usr/bin/qemu-alpha-static as binfmt interpreter for alpha
Setting /usr/bin/qemu-arm-static as binfmt interpreter for arm
Setting /usr/bin/qemu-armeb-static as binfmt interpreter for armeb
Setting /usr/bin/qemu-sparc-static as binfmt interpreter for sparc
Setting /usr/bin/qemu-sparc32plus-static as binfmt interpreter for sparc32plus
Setting /usr/bin/qemu-sparc64-static as binfmt interpreter for sparc64
Setting /usr/bin/qemu-ppc-static as binfmt interpreter for ppc
Setting /usr/bin/qemu-ppc64-static as binfmt interpreter for ppc64
Setting /usr/bin/qemu-ppc64le-static as binfmt interpreter for ppc64le
Setting /usr/bin/qemu-m68k-static as binfmt interpreter for m68k
Setting /usr/bin/qemu-mips-static as binfmt interpreter for mips
Setting /usr/bin/qemu-mipsel-static as binfmt interpreter for mipsel
Setting /usr/bin/qemu-mipsn32-static as binfmt interpreter for mipsn32
Setting /usr/bin/qemu-mipsn32el-static as binfmt interpreter for mipsn32el
Setting /usr/bin/qemu-mips64-static as binfmt interpreter for mips64
Setting /usr/bin/qemu-mips64el-static as binfmt interpreter for mips64el
Setting /usr/bin/qemu-sh4-static as binfmt interpreter for sh4
Setting /usr/bin/qemu-sh4eb-static as binfmt interpreter for sh4eb
Setting /usr/bin/qemu-s390x-static as binfmt interpreter for s390x
Setting /usr/bin/qemu-aarch64-static as binfmt interpreter for aarch64
Setting /usr/bin/qemu-aarch64_be-static as binfmt interpreter for aarch64_be
Setting /usr/bin/qemu-hppa-static as binfmt interpreter for hppa
Setting /usr/bin/qemu-riscv32-static as binfmt interpreter for riscv32
Setting /usr/bin/qemu-riscv64-static as binfmt interpreter for riscv64
Setting /usr/bin/qemu-xtensa-static as binfmt interpreter for xtensa
Setting /usr/bin/qemu-xtensaeb-static as binfmt interpreter for xtensaeb
Setting /usr/bin/qemu-microblaze-static as binfmt interpreter for microblaze
Setting /usr/bin/qemu-microblazeel-static as binfmt interpreter for microblazeel
Setting /usr/bin/qemu-or1k-static as binfmt interpreter for or1k
Setting /usr/bin/qemu-hexagon-static as binfmt interpreter for hexagon
docker buildx create --use
gallant_khorana
docker buildx build --platform linux/amd64,linux/arm64 -t gcr.io/k8s-staging-npd/node-problem-detector:v0.8.13-4-g9fd58e3 --build-arg BASEIMAGE=registry.k8s.io/build-image/debian-base:bullseye-v1.4.3 --build-arg LOGCOUNTER=./bin/log-counter .
WARNING: No output specified with docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load
[+] Building 435.6s (36/36) FINISHED
 => [internal] booting buildkit                                                                                                                           1.6s
 => => pulling image moby/buildkit:buildx-stable-1                                                                                                        1.0s
 => => creating container buildx_buildkit_gallant_khorana0                                                                                                0.6s
 => [internal] load .dockerignore                                                                                                                         0.0s
 => => transferring context: 2B                                                                                                                           0.0s
 => [internal] load build definition from Dockerfile                                                                                                      0.0s
 => => transferring dockerfile: 1.94kB                                                                                                                    0.0s
 => [linux/amd64 internal] load metadata for registry.k8s.io/build-image/debian-base:bullseye-v1.4.3                                                      2.2s
 => [linux/amd64 internal] load metadata for docker.io/library/golang:1.20.0@sha256:f69d47fedd3b2ebd23bcf473c0b78522ebbc1823f06b7d47f45f04a30bdc901d      2.4s
 => [linux/arm64 internal] load metadata for registry.k8s.io/build-image/debian-base:bullseye-v1.4.3                                                      2.2s
 => [linux/arm64 internal] load metadata for docker.io/library/golang:1.20.0@sha256:f69d47fedd3b2ebd23bcf473c0b78522ebbc1823f06b7d47f45f04a30bdc901d      2.5s
 => [internal] load build context                                                                                                                         1.5s
 => => transferring context: 60.79MB                                                                                                                      1.4s
 => [linux/arm64 stage-1 1/6] FROM registry.k8s.io/build-image/debian-base:bullseye-v1.4.3@sha256:3fcf9bcc4949392155cde1f96591ba3eda092f4735a9cc856bf526  1.7s
 => => resolve registry.k8s.io/build-image/debian-base:bullseye-v1.4.3@sha256:3fcf9bcc4949392155cde1f96591ba3eda092f4735a9cc856bf526d71f48af47            0.0s
 => => sha256:f91cc52e2c97c480189bb2da87653dd9abfe2679c36fc41b9db68878827748b4 25.32MB / 25.32MB                                                          0.6s
 => => extracting sha256:f91cc52e2c97c480189bb2da87653dd9abfe2679c36fc41b9db68878827748b4                                                                 1.2s
 => [linux/arm64 builder 1/8] FROM docker.io/library/golang:1.20.0@sha256:f69d47fedd3b2ebd23bcf473c0b78522ebbc1823f06b7d47f45f04a30bdc901d               12.2s
 => => resolve docker.io/library/golang:1.20.0@sha256:f69d47fedd3b2ebd23bcf473c0b78522ebbc1823f06b7d47f45f04a30bdc901d                                    0.0s
 => => sha256:77fe3ac745a5ff347cfde07c7e36e684d71ada78b8efa592ba32dcd423a2ac32 155B / 155B                                                                0.2s
 => => sha256:fb91627ff2638784eb011caf304c02690a22167a783fa6cb17324bc320f4c878 95.20MB / 95.20MB                                                          1.4s
 => => sha256:4bfb8dc93d4197860c2bff47f2c2f280c2dd8ed699e7b3241aa325ecee53c7d7 54.68MB / 54.68MB                                                          0.9s
 => => sha256:530999854f4629231859226fd635c5088244ca7736053631fe412d6325352e6e 81.39MB / 81.39MB                                                          1.2s
 => => sha256:0855378f8903bde22cfbcee08cd239678716cf01f24a3fca9478ef4121a84d91 10.87MB / 10.87MB                                                          0.5s
 => => sha256:7b716680367d1dac0e54c48f75506323e0bb03628542a0fd6db39efeeee9adf5 5.15MB / 5.15MB                                                            0.2s
 => => sha256:c345c9e441f5f49235768af74b8ab37743652d38958afaa000edd56d7b2f0540 53.68MB / 53.68MB                                                          0.6s
 => => extracting sha256:c345c9e441f5f49235768af74b8ab37743652d38958afaa000edd56d7b2f0540                                                                 1.8s
 => => extracting sha256:7b716680367d1dac0e54c48f75506323e0bb03628542a0fd6db39efeeee9adf5                                                                 0.2s
 => => extracting sha256:0855378f8903bde22cfbcee08cd239678716cf01f24a3fca9478ef4121a84d91                                                                 0.2s
 => => extracting sha256:4bfb8dc93d4197860c2bff47f2c2f280c2dd8ed699e7b3241aa325ecee53c7d7                                                                 2.1s
 => => extracting sha256:530999854f4629231859226fd635c5088244ca7736053631fe412d6325352e6e                                                                 2.3s
 => => extracting sha256:fb91627ff2638784eb011caf304c02690a22167a783fa6cb17324bc320f4c878                                                                 3.9s
 => => extracting sha256:77fe3ac745a5ff347cfde07c7e36e684d71ada78b8efa592ba32dcd423a2ac32                                                                 0.0s
 => [linux/amd64 builder 1/8] FROM docker.io/library/golang:1.20.0@sha256:f69d47fedd3b2ebd23bcf473c0b78522ebbc1823f06b7d47f45f04a30bdc901d               13.8s
 => => resolve docker.io/library/golang:1.20.0@sha256:f69d47fedd3b2ebd23bcf473c0b78522ebbc1823f06b7d47f45f04a30bdc901d                                    0.0s
 => => sha256:24c88dfe5f5b0e9c19438c407c7e04a10b0486ead845e102dab55d79c6e56cce 156B / 156B                                                                0.2s
 => => sha256:9a8ca04f4e787c557c6b220937a4252770e348b9374e6ac222c54f2d7bfb4dcf 99.87MB / 99.87MB                                                          1.2s
 => => sha256:bfcb68b5bd105d3f88a2c15354cff6c253bedc41d83c1da28b3d686c37cd9103 85.98MB / 85.98MB                                                          1.1s
 => => sha256:9bd150679dbdb02d9d4df4457d54211d6ee719ca7bc77747a7be4cd99ae03988 54.58MB / 54.58MB                                                          0.7s
 => => sha256:56261d0e6b05ece42650b14830960db5b42a9f23479d868256f91d96869ac0c2 10.88MB / 10.88MB                                                          0.3s
 => => sha256:f049f75f014ee8fec2d4728b203c9cbee0502ce142aec030f874aa28359e25f1 5.16MB / 5.16MB                                                            0.2s
 => => sha256:bbeef03cda1f5d6c9e20c310c1c91382a6b0a1a2501c3436b28152f13896f082 55.03MB / 55.03MB                                                          0.7s
 => => extracting sha256:bbeef03cda1f5d6c9e20c310c1c91382a6b0a1a2501c3436b28152f13896f082                                                                 1.8s
 => => extracting sha256:f049f75f014ee8fec2d4728b203c9cbee0502ce142aec030f874aa28359e25f1                                                                 0.2s
 => => extracting sha256:56261d0e6b05ece42650b14830960db5b42a9f23479d868256f91d96869ac0c2                                                                 0.2s
 => => extracting sha256:9bd150679dbdb02d9d4df4457d54211d6ee719ca7bc77747a7be4cd99ae03988                                                                 2.2s
 => => extracting sha256:bfcb68b5bd105d3f88a2c15354cff6c253bedc41d83c1da28b3d686c37cd9103                                                                 2.8s
 => => extracting sha256:9a8ca04f4e787c557c6b220937a4252770e348b9374e6ac222c54f2d7bfb4dcf                                                                 3.7s
 => => extracting sha256:24c88dfe5f5b0e9c19438c407c7e04a10b0486ead845e102dab55d79c6e56cce                                                                 0.0s
 => [linux/amd64 stage-1 1/6] FROM registry.k8s.io/build-image/debian-base:bullseye-v1.4.3@sha256:3fcf9bcc4949392155cde1f96591ba3eda092f4735a9cc856bf526  1.7s
 => => resolve registry.k8s.io/build-image/debian-base:bullseye-v1.4.3@sha256:3fcf9bcc4949392155cde1f96591ba3eda092f4735a9cc856bf526d71f48af47            0.0s
 => => sha256:c4e25f9975ade78ed0838b6cbf3f4fe1c6bf817811f3c8e676b1cbb6a7104950 25.44MB / 25.44MB                                                          0.5s
 => => extracting sha256:c4e25f9975ade78ed0838b6cbf3f4fe1c6bf817811f3c8e676b1cbb6a7104950                                                                 1.2s
 => [linux/amd64 stage-1 2/6] RUN clean-install util-linux bash libsystemd-dev                                                                            8.4s
 => [linux/arm64 stage-1 2/6] RUN clean-install util-linux bash libsystemd-dev                                                                           42.1s
 => [linux/amd64 stage-1 3/6] RUN test -h /etc/localtime && rm -f /etc/localtime && cp /usr/share/zoneinfo/UTC /etc/localtime || true                     0.2s
 => [linux/arm64 builder 2/8] RUN apt-get update --fix-missing && apt-get --yes install libsystemd-dev gcc-aarch64-linux-gnu                             30.6s
 => [linux/amd64 builder 2/8] RUN apt-get update --fix-missing && apt-get --yes install libsystemd-dev gcc-aarch64-linux-gnu                             28.5s
 => [linux/amd64 builder 3/8] RUN go version                                                                                                              0.1s
 => [linux/amd64 builder 4/8] RUN go install github.com/tools/godep@latest                                                                               15.8s
 => [linux/arm64 builder 3/8] RUN go version                                                                                                              0.4s
 => [linux/arm64 builder 4/8] RUN go install github.com/tools/godep@latest                                                                              153.6s
 => [linux/arm64 stage-1 3/6] RUN test -h /etc/localtime && rm -f /etc/localtime && cp /usr/share/zoneinfo/UTC /etc/localtime || true                     0.2s
 => [linux/amd64 builder 5/8] RUN godep version                                                                                                           0.2s
 => [linux/amd64 builder 6/8] COPY . /gopath/src/k8s.io/node-problem-detector/                                                                            0.5s
 => [linux/amd64 builder 7/8] WORKDIR /gopath/src/k8s.io/node-problem-detector                                                                            0.0s
 => [linux/amd64 builder 8/8] RUN make bin/node-problem-detector bin/health-checker bin/log-counter                                                      23.9s
 => [linux/amd64 stage-1 4/6] COPY --from=builder /gopath/src/k8s.io/node-problem-detector/bin/node-problem-detector /node-problem-detector               0.0s
 => [linux/amd64 stage-1 5/6] COPY --from=builder /gopath/src/k8s.io/node-problem-detector/bin/health-checker /gopath/src/k8s.io/node-problem-detector/.  0.0s
 => [linux/amd64 stage-1 6/6] COPY --from=builder /gopath/src/k8s.io/node-problem-detector/config/ /config                                                0.0s
 => [linux/arm64 builder 5/8] RUN godep version                                                                                                           0.6s
 => [linux/arm64 builder 6/8] COPY . /gopath/src/k8s.io/node-problem-detector/                                                                            0.4s
 => [linux/arm64 builder 7/8] WORKDIR /gopath/src/k8s.io/node-problem-detector                                                                            0.0s
 => [linux/arm64 builder 8/8] RUN make bin/node-problem-detector bin/health-checker bin/log-counter                                                     233.2s
 => [linux/arm64 stage-1 4/6] COPY --from=builder /gopath/src/k8s.io/node-problem-detector/bin/node-problem-detector /node-problem-detector               0.0s
 => [linux/arm64 stage-1 5/6] COPY --from=builder /gopath/src/k8s.io/node-problem-detector/bin/health-checker /gopath/src/k8s.io/node-problem-detector/.  0.0s
 => [linux/arm64 stage-1 6/6] COPY --from=builder /gopath/src/k8s.io/node-problem-detector/config/ /config                                                0.0s
CGO_ENABLED=1 GOOS=linux GO111MODULE=on go build \
	-mod vendor \
	-o bin/node-problem-detector \
	-ldflags '-X k8s.io/node-problem-detector/pkg/version.version=v0.8.13-4-g9fd58e3' \
	-tags "journald " \
	./cmd/nodeproblemdetector
# github.com/coreos/go-systemd/sdjournal
vendor/github.com/coreos/go-systemd/sdjournal/journal.go:27:11: fatal error: systemd/sd-journal.h: No such file or directory
   27 | // #include <systemd/sd-journal.h>
      |           ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:199: bin/node-problem-detector] Error 2
```

**Comment by smileusd** [comment]
Another issue is that any arm base image for multiarch/qemu-user-static ? I want to make that on my MAC M2 machine
```docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
exec /register: exec format error
docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
exec /register: exec format error

```

**Comment by smileusd** [comment]
@Random-Liu PTAL

**Comment by btiernay** [comment]
I get a similar issue on mac on af2226183ff
```
WARNING: No output specified with docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load
CGO_ENABLED=1 GOOS=linux GO111MODULE=on go build \
		-mod vendor \
		-o bin/node-problem-detector \
		-ldflags '-X k8s.io/node-problem-detector/pkg/version.version=v0.8.13-15-gaf22261' \
		-tags "journald " \
		./cmd/nodeproblemdetector
# runtime/cgo
linux_syscall.c:67:13: error: call to undeclared function 'setresgid'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
linux_syscall.c:67:13: note: did you mean 'setregid'?
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/unistd.h:593:6: note: 'setregid' declared here
linux_syscall.c:73:13: error: call to undeclared function 'setresuid'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
linux_syscall.c:73:13: note: did you mean 'setreuid'?
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/unistd.h:595:6: note: 'setreuid' declared here
make: *** [bin/node-problem-detector] Error 1
```

```
uname -a
Darwin C02ZK5VPMD6P 22.5.0 Darwin Kernel Version 22.5.0: Mon Apr 24 20:51:50 PDT 2023; root:xnu-8796.121.2~5/RELEASE_X86_64 x86_64 i386 Darwin
```
```
xcode-select --version
xcode-select version 2397.
```
<details><summary>Details</summary>
<p>
```bash
make
rm -rf bin/
rm -rf test/bin/
rm -f node-problem-detector-*.tar.gz*
rm -rf output/
rm -f coverage.out
docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
Setting /usr/bin/qemu-alpha-static as binfmt interpreter for alpha
Setting /usr/bin/qemu-arm-static as binfmt interpreter for arm
Setting /usr/bin/qemu-armeb-static as binfmt interpreter for armeb
Setting /usr/bin/qemu-sparc-static as binfmt interpreter for sparc
Setting /usr/bin/qemu-sparc32plus-static as binfmt interpreter for sparc32plus
Setting /usr/bin/qemu-sparc64-static as binfmt interpreter for sparc64
Setting /usr/bin/qemu-ppc-static as binfmt interpreter for ppc
Setting /usr/bin/qemu-ppc64-static as binfmt interpreter for ppc64
Setting /usr/bin/qemu-ppc64le-static as binfmt interpreter for ppc64le
Setting /usr/bin/qemu-m68k-static as binfmt interpreter for m68k
Setting /usr/bin/qemu-mips-static as binfmt interpreter for mips
Setting /usr/bin/qemu-mipsel-static as binfmt interpreter for mipsel
Setting /usr/bin/qemu-mipsn32-static as binfmt interpreter for mipsn32
Setting /usr/bin/qemu-mipsn32el-static as binfmt interpreter for mipsn32el
Setting /usr/bin/qemu-mips64-static as binfmt interpreter for mips64
Setting /usr/bin/qemu-mips64el-static as binfmt interpreter for mips64el
Setting /usr/bin/qemu-sh4-static as binfmt interpreter for sh4
Setting /usr/bin/qemu-sh4eb-static as binfmt interpreter for sh4eb
Setting /usr/bin/qemu-s390x-static as binfmt interpreter for s390x
Setting /usr/bin/qemu-aarch64-static as binfmt interpreter for aarch64
Setting /usr/bin/qemu-aarch64_be-static as binfmt interpreter for aarch64_be
Setting /usr/bin/qemu-hppa-static as binfmt interpreter for hppa
Setting /usr/bin/qemu-riscv32-static as binfmt interpreter for riscv32
Setting /usr/bin/qemu-riscv64-static as binfmt interpreter for riscv64
Setting /usr/bin/qemu-xtensa-static as binfmt interpreter for xtensa
Setting /usr/bin/qemu-xtensaeb-static as binfmt interpreter for xtensaeb
Setting /usr/bin/qemu-microblaze-static as binfmt interpreter for microblaze
Setting /usr/bin/qemu-microblazeel-static as binfmt interpreter for microblazeel
Setting /usr/bin/qemu-or1k-static as binfmt interpreter for or1k
Setting /usr/bin/qemu-hexagon-static as binfmt interpreter for hexagon
docker buildx create --use
elastic_rhodes
docker buildx build --platform linux/amd64,linux/arm64 -t gcr.io/k8s-staging-npd/node-problem-detector:v0.8.13-15-gaf22261 --build-arg BASEIMAGE=registry.k8s.io/build-image/debian-base:bullseye-v1.4.3 --build-arg LOGCOUNTER=./bin/log-counter .
[+] Building 530.6s (33/33) FINISHED
 => [internal] booting buildkit                                                                                                                                                           1.9s
 => => pulling image moby/buildkit:buildx-stable-1                                                                                                                                        1.2s
 => => creating container buildx_buildkit_elastic_rhodes0                                                                                                                                 0.7s
 => [internal] load build definition from Dockerfile                                                                                                                                      0.1s
 => => transferring dockerfile: 1.88kB                                                                                                                                                    0.0s
 => [internal] load .dockerignore                                                                                                                                                         0.1s
 => => transferring context: 2B                                                                                                                                                           0.0s
 => [linux/amd64 internal] load metadata for docker.io/library/golang:1.20.3@sha256:bcc311ec9655c350df3899611fdf134806f97a3e3b2c06c2b5c0696428503814                                      3.1s
 => [linux/amd64 internal] load metadata for registry.k8s.io/build-image/debian-base:bullseye-v1.4.3                                                                                      1.9s
 => [linux/arm64 internal] load metadata for docker.io/library/golang:1.20.3@sha256:bcc311ec9655c350df3899611fdf134806f97a3e3b2c06c2b5c0696428503814                                      3.1s
 => [linux/arm64 internal] load metadata for registry.k8s.io/build-image/debian-base:bullseye-v1.4.3                                                                                      1.8s
 => [auth] library/golang:pull token for registry-1.docker.io                                                                                                                             0.0s
 => [linux/amd64 builder 1/6] FROM docker.io/library/golang:1.20.3@sha256:bcc311ec9655c350df3899611fdf134806f97a3e3b2c06c2b5c0696428503814                                              309.5s
 => => resolve docker.io/library/golang:1.20.3@sha256:bcc311ec9655c350df3899611fdf134806f97a3e3b2c06c2b5c0696428503814                                                                    0.0s
 => => sha256:5997e4205ef719aca0e749aaed2000f49a2485a8f3f6861ad9582f90395754c5 155B / 155B                                                                                                0.1s
 => => sha256:92ac76a462cb8d40a876912d250348f391e1ebf2ff2366684011a164d8480456 100.14MB / 100.14MB                                                                                      303.5s
 => => sha256:db119aaf144b2bc25823dcfc08c6e276826608564a4fb7e50c46d6fc8b8dea14 85.99MB / 85.99MB                                                                                        281.0s
 => => sha256:5a7a2c95f0f8b221d776ccf35911b68eec2cf9414a44d216205a6f03e381d3d7 54.58MB / 54.58MB                                                                                        210.6s
 => => sha256:670730c27c2eacf07897a6e94fe55423ea50b884d9c28161a283bbbf064d1124 10.88MB / 10.88MB                                                                                        109.0s
 => => sha256:68a71c865a2c34678c6dea55e4b0928f751ee3c0ca91cace6e4e0578c534d6cf 5.17MB / 5.17MB                                                                                           19.5s
 => => sha256:3e440a7045683e27f8e2fa04000e0e078d8dfac0c971358ae0f8c65c13321c8e 55.05MB / 55.05MB                                                                                        168.7s
 => => extracting sha256:3e440a7045683e27f8e2fa04000e0e078d8dfac0c971358ae0f8c65c13321c8e                                                                                                 1.7s
 => => extracting sha256:68a71c865a2c34678c6dea55e4b0928f751ee3c0ca91cace6e4e0578c534d6cf                                                                                                 0.2s
 => => extracting sha256:670730c27c2eacf07897a6e94fe55423ea50b884d9c28161a283bbbf064d1124                                                                                                 0.2s
 => => extracting sha256:5a7a2c95f0f8b221d776ccf35911b68eec2cf9414a44d216205a6f03e381d3d7                                                                                                 2.2s
 => => extracting sha256:db119aaf144b2bc25823dcfc08c6e276826608564a4fb7e50c46d6fc8b8dea14                                                                                                 3.3s
 => => extracting sha256:92ac76a462cb8d40a876912d250348f391e1ebf2ff2366684011a164d8480456                                                                                                 4.2s
 => => extracting sha256:5997e4205ef719aca0e749aaed2000f49a2485a8f3f6861ad9582f90395754c5                                                                                                 0.0s
 => [linux/arm64 builder 1/6] FROM docker.io/library/golang:1.20.3@sha256:bcc311ec9655c350df3899611fdf134806f97a3e3b2c06c2b5c0696428503814                                                0.0s
 => => resolve docker.io/library/golang:1.20.3@sha256:bcc311ec9655c350df3899611fdf134806f97a3e3b2c06c2b5c0696428503814                                                                    0.0s
 => [linux/arm64 stage-1 1/6] FROM registry.k8s.io/build-image/debian-base:bullseye-v1.4.3@sha256:3fcf9bcc4949392155cde1f96591ba3eda092f4735a9cc856bf526d71f48af47                      232.7s
 => => resolve registry.k8s.io/build-image/debian-base:bullseye-v1.4.3@sha256:3fcf9bcc4949392155cde1f96591ba3eda092f4735a9cc856bf526d71f48af47                                            0.0s
 => => sha256:f91cc52e2c97c480189bb2da87653dd9abfe2679c36fc41b9db68878827748b4 25.32MB / 25.32MB                                                                                        231.8s
 => => extracting sha256:f91cc52e2c97c480189bb2da87653dd9abfe2679c36fc41b9db68878827748b4                                                                                                 0.8s
 => [internal] load build context                                                                                                                                                         3.3s
 => => transferring context: 61.40MB                                                                                                                                                      3.2s
 => [linux/amd64 stage-1 1/6] FROM registry.k8s.io/build-image/debian-base:bullseye-v1.4.3@sha256:3fcf9bcc4949392155cde1f96591ba3eda092f4735a9cc856bf526d71f48af47                      136.9s
 => => resolve registry.k8s.io/build-image/debian-base:bullseye-v1.4.3@sha256:3fcf9bcc4949392155cde1f96591ba3eda092f4735a9cc856bf526d71f48af47                                            0.0s
 => => sha256:c4e25f9975ade78ed0838b6cbf3f4fe1c6bf817811f3c8e676b1cbb6a7104950 25.44MB / 25.44MB                                                                                        135.8s
 => => extracting sha256:c4e25f9975ade78ed0838b6cbf3f4fe1c6bf817811f3c8e676b1cbb6a7104950                                                                                                 0.9s
 => [linux/amd64 stage-1 2/6] RUN clean-install util-linux bash libsystemd-dev                                                                                                           15.6s
 => [linux/amd64 stage-1 3/6] RUN test -h /etc/localtime && rm -f /etc/localtime && cp /usr/share/zoneinfo/UTC /etc/localtime || true                                                     0.1s
 => [linux/arm64 stage-1 2/6] RUN clean-install util-linux bash libsystemd-dev                                                                                                           59.9s
 => [linux/arm64 stage-1 3/6] RUN test -h /etc/localtime && rm -f /etc/localtime && cp /usr/share/zoneinfo/UTC /etc/localtime || true                                                     0.1s
 => [linux/amd64 builder 2/6] RUN apt-get update --fix-missing && apt-get --yes install libsystemd-dev gcc-aarch64-linux-gnu                                                             90.9s
 => [linux/arm64 builder 2/6] RUN apt-get update --fix-missing && apt-get --yes install libsystemd-dev gcc-aarch64-linux-gnu                                                             97.4s
 => [linux/amd64 builder 3/6] RUN go version                                                                                                                                              0.1s
 => [linux/amd64 builder 4/6] COPY . /gopath/src/k8s.io/node-problem-detector/                                                                                                            1.3s
 => [linux/amd64 builder 5/6] WORKDIR /gopath/src/k8s.io/node-problem-detector                                                                                                            0.0s
 => [linux/amd64 builder 6/6] RUN make bin/node-problem-detector bin/health-checker bin/log-counter                                                                                     119.8s
 => [linux/arm64 builder 3/6] RUN go version                                                                                                                                              0.1s
 => [linux/arm64 builder 4/6] COPY . /gopath/src/k8s.io/node-problem-detector/                                                                                                            0.5s
 => [linux/arm64 builder 5/6] WORKDIR /gopath/src/k8s.io/node-problem-detector                                                                                                            0.1s
 => [linux/arm64 builder 6/6] RUN make bin/node-problem-detector bin/health-checker bin/log-counter                                                                                     117.2s
 => [linux/amd64 stage-1 4/6] COPY --from=builder /gopath/src/k8s.io/node-problem-detector/bin/node-problem-detector /node-problem-detector                                               0.1s
 => [linux/amd64 stage-1 5/6] COPY --from=builder /gopath/src/k8s.io/node-problem-detector/bin/health-checker /gopath/src/k8s.io/node-problem-detector/./bin/log-counter /home/kubernete  0.1s
 => [linux/amd64 stage-1 6/6] COPY --from=builder /gopath/src/k8s.io/node-problem-detector/config/ /config                                                                                0.0s
 => [linux/arm64 stage-1 4/6] COPY --from=builder /gopath/src/k8s.io/node-problem-detector/bin/node-problem-detector /node-problem-detector                                               0.1s
 => [linux/arm64 stage-1 5/6] COPY --from=builder /gopath/src/k8s.io/node-problem-detector/bin/health-checker /gopath/src/k8s.io/node-problem-detector/./bin/log-counter /home/kubernete  0.1s
 => [linux/arm64 stage-1 6/6] COPY --from=builder /gopath/src/k8s.io/node-problem-detector/config/ /config                                                                                0.0s
WARNING: No output specified with docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load
CGO_ENABLED=1 GOOS=linux GO111MODULE=on go build \
		-mod vendor \
		-o bin/node-problem-detector \
		-ldflags '-X k8s.io/node-problem-detector/pkg/version.version=v0.8.13-15-gaf22261' \
		-tags "journald " \
		./cmd/nodeproblemdetector
# runtime/cgo
linux_syscall.c:67:13: error: call to undeclared function 'setresgid'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
linux_syscall.c:67:13: note: did you mean 'setregid'?
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/unistd.h:593:6: note: 'setregid' declared here
linux_syscall.c:73:13: error: call to undeclared function 'setresuid'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
linux_syscall.c:73:13: note: did you mean 'setreuid'?
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/unistd.h:595:6: note: 'setreuid' declared here
make: *** [bin/node-problem-detector] Error 1
```
</p>
</details>

**Comment by btiernay** [comment]
Looks like it's `cgo` per https://github.com/golang/go/issues/22510#issuecomment-340938955. This works:

```bash
ENABLE_JOURNALD=0 make
```

But obviously that monitor won't be built and linked.

**Comment by btiernay** [comment]
Looks like there might be an issue with the `Makefile` variable assigments.  Suggestion is to replace:

```make
ifeq ($(go env GOHOSTOS), darwin)
ENABLE_JOURNALD=0
else ifeq ($(go env GOHOSTOS), windows)
ENABLE_JOURNALD=0
endif
```
with
```make
ifeq ($(shell go env GOHOSTOS), darwin)
ENABLE_JOURNALD=0
else ifeq ($(shell go env GOHOSTOS), windows)
ENABLE_JOURNALD=0
endif
```

**Comment by hakman** [comment]
Looks like this was fixed by #763.
/close
